benches/oneshot/ipa.rs

```diff
@@ -13,36 +13,33 @@ use std::time::Instant;
 async fn main() -> Result<(), Error> {
     const MAX_BREAKDOWN_KEY: u32 = 16;
     const MAX_TRIGGER_VALUE: u32 = 5;
-    const MAX_QUERY_SIZE: usize = 1000;
-    const NUM_USERS: usize = 300;
+    const QUERY_SIZE: usize = 100;
     const MAX_RECORDS_PER_USER: usize = 10;
 
     let mut config = TestWorldConfig::default();
-    config.gateway_config =
-        GatewayConfig::symmetric_buffers((NUM_USERS * MAX_RECORDS_PER_USER).clamp(16, 1024));
+    config.gateway_config = GatewayConfig::symmetric_buffers(QUERY_SIZE.clamp(16, 1024));
 
     let random_seed = thread_rng().gen();
     println!("Using random seed: {random_seed}");
     let mut rng = StdRng::seed_from_u64(random_seed);
 
     let mut total_count = 0;
 
-    let mut random_user_records = Vec::with_capacity(NUM_USERS);
-    while random_user_records.len() < NUM_USERS && total_count < MAX_QUERY_SIZE {
+    let mut random_user_records = Vec::with_capacity(QUERY_SIZE / MAX_RECORDS_PER_USER);
+    while total_count < QUERY_SIZE {
         let mut records_for_user = generate_random_user_records_in_reverse_chronological_order(
             &mut rng,
             MAX_RECORDS_PER_USER,
             MAX_BREAKDOWN_KEY,
             MAX_TRIGGER_VALUE,
         );
-        if records_for_user.len() > MAX_QUERY_SIZE - total_count {
-            records_for_user.truncate(MAX_QUERY_SIZE - total_count);
+        if records_for_user.len() > QUERY_SIZE - total_count {
+            records_for_user.truncate(QUERY_SIZE - total_count);
         }
```

Comment:
You can just truncate unconditionally.

```suggestion
        records_for_user.truncate(QUERY_SIZE - total_count);
```